maths/factorial.dart

```diff
@@ -1,7 +1,11 @@
 void main() {
   var n = 5;
-  var fac = factorial(n);
+  if(n>=0 && n.runtimeType == int){
```

Comment:
Instead of checking types, Why not replace `var n =5;` with `int n=5;`
This will help us avoid the unnecessary type checking.

Reply:
We can do that, I would just say to leave it a variable in case if user input is needed. Whatever you want though.

Reply:
Since we are not taking any user input and dart has a great types. let's exploit that. 

you can add test for the same.